src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,589 @@
+        asserts.assert_equal(result[0].Status, Status.Success, "ACL write failed")
+
+    async def get_descriptor_server_list(self, ctrl):
+        return await self.read_single_attribute_check_success(
+            endpoint=0,
+            dev_ctrl=ctrl,
+            cluster=Clusters.Descriptor,
+            attribute=Clusters.Descriptor.Attributes.ServerList
+        )
+
+    async def get_idle_mode_duration(self, ctrl):
+        return await self.read_single_attribute_check_success(
+            endpoint=0,
+            dev_ctrl=ctrl,
+            cluster=Clusters.IcdManagement,
+            attribute=Clusters.IcdManagement.Attributes.IdleModeDuration
+        )
+
+    def verify_attribute_data(self, sub, cluster, attribute, ep=0):
+        sub_attrs = sub
+        if isinstance(sub, Clusters.Attribute.SubscriptionTransaction):
+            sub_attrs = sub.GetAttributes()
+
+        asserts.assert_true(ep in sub_attrs, "Must have read endpoint %s data" % ep)
+        asserts.assert_true(cluster in sub_attrs[0], "Must have read %s cluster data" % cluster.__name__)
+        asserts.assert_true(attribute in sub_attrs[0][cluster],
+                            "Must have read back attribute %s" % attribute.__name__)
+
+    def get_attribute_from_sub_dict(self, sub, cluster, attribute, ep=0):
+        return sub[ep][cluster][attribute]
+
+    def get_typed_attribute_path(self, attribute, ep=0):
+        return TypedAttributePath(
+            Path=AttributePath(
+                EndpointId=ep,
+                Attribute=attribute
+            )
+        )
+
+    async def get_dut_acl(self):
+        sub = await self.default_controller.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=[(0, Clusters.AccessControl.Attributes.Acl)],
+            keepSubscriptions=False,
+            fabricFiltered=True
+        )
+
+        acl_list = self.get_attribute_from_sub_dict(
+            sub=sub,
+            cluster=Clusters.AccessControl,
+            attribute=Clusters.AccessControl.Attributes.Acl
+        )
+
+        return acl_list
+
+    def is_uint32(self, var):
+        return isinstance(var, int) and 0 <= var <= 4294967295
+
+    def is_uint16(self, var):
+        return isinstance(var, int) and 0 <= var <= 65535
+
+    @async_test_body
+    async def test_TC_IDM_4_2(self):
+
+        # Test setup
+        cluster_rev_attr = Clusters.BasicInformation.Attributes.ClusterRevision
+        cluster_rev_attr_typed_path = self.get_typed_attribute_path(cluster_rev_attr)
+        node_label_attr = Clusters.BasicInformation.Attributes.NodeLabel
+        node_label_attr_path = [(0, node_label_attr)]
+        node_label_attr_typed_path = self.get_typed_attribute_path(node_label_attr)
+        SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = 0
+        INVALID_ACTION_ERROR_CODE = 0x580
+        same_min_max_interval_sec = 3
```

Comment:
This seems like a constant. should it be `SAME_MIN_MAX_INTERVAL_SEC` ?

Reply:
In this case 3 is an arbitrary value, it doesn't hold any specific meaning other than only waiting 3 seconds to read an updated attribute value for test case brevity. It's only used in one of the steps.

I will move it to the step 8 block where it's used so its purpose is clearer.